src/main.rs

```diff
@@ -84,6 +38,27 @@ fn bind_socket(addr: SocketAddr) -> UdpSocket {
     socket
 }
 
+/// Run a future with the runtime specified by config.
+fn run<F>(future: F, threads_count: usize)
+    where F: Future<Item = (), Error = Error> + Send + 'static
+{
+    if threads_count == 1 {
+        let mut runtime = runtime::current_thread::Runtime::new().expect("Failed to create runtime");
+        runtime.block_on(future).expect("Execution was terminated with error");
+    } else {
+        let mut threadpool_builder = thread_pool::Builder::new();
+        threadpool_builder.name_prefix("tox-node-thread-");
+        if threads_count != 0 {
+            threadpool_builder.pool_size(threads_count);
```

Comment:
Did you check that it spawned actually threads_count?

Reply:
I didn't, but I checked that CLI argument is passed here and that just calling without using result should be fine: https://docs.rs/tokio-threadpool/0.1.4/src/tokio_threadpool/builder.rs.html#140-146